toolchain/check/handle_let_and_var.cpp

```diff
@@ -63,7 +64,7 @@ static auto GetOrAddStorage(Context& context, SemIR::InstId pattern_id)
         context.entity_names().Get(binding_pattern->entity_name_id).name_id;
   }
   return context.AddInst(SemIR::LocIdAndInst::UncheckedLoc(
-      pattern.loc_id, SemIR::VarStorage{.type_id = subpattern.type_id(),
+      pattern.loc_id, SemIR::VarStorage{.type_id = pattern.inst.type_id(),
```

Comment:
What is the difference between the pattern and subpattern types? (Or, what fails/breaks without this?)

Reply:
Hmm. I don't remember why I changed this; it works either way, because the two types are always the same. Looking at it now, though, the revised version seems marginally better to me -- if there ever were a way for a `var` pattern to have a different type than its subpattern, it seems like the storage should have the `var` pattern's type, not the subpattern's type.